packages/cli/src/zosjobs/submit/Submit.shared.handler.ts

```diff
@@ -31,150 +31,158 @@ export default class SharedSubmitHandler extends ZosmfBaseHandler {
+                            jclSymbols: this.mArguments.jclSymbols});
+                        apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
+                        source = this.mArguments.dataset;
+
+                        if (parms.viewAllSpoolContent) {
+                            spoolFilesResponse = apiObj;
+                        }
                     }
 
                     break;
-                } else {
-                    response = await SubmitJobs.submitJobCommon(this.mSession, {jobDataSet: this.mArguments.dataset,
+                // Submit JCL from a USS file
+                case "uss-file":
+                    response = await SubmitJobs.submitJobCommon(this.mSession, {jobUSSFile: this.mArguments.file,
                         jclSymbols: this.mArguments.jclSymbols});
                     apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
-                    source = this.mArguments.dataset;
+                    source = this.mArguments.ussfile;
 
                     if (parms.viewAllSpoolContent) {
                         spoolFilesResponse = apiObj;
                     }
-                }
 
-                break;
-            // Submit JCL from a USS file
-            case "uss-file":
-                response = await SubmitJobs.submitJobCommon(this.mSession, {jobUSSFile: this.mArguments.file,
-                    jclSymbols: this.mArguments.jclSymbols});
-                apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
-                source = this.mArguments.ussfile;
-
-                if (parms.viewAllSpoolContent) {
-                    spoolFilesResponse = apiObj;
-                }
-
-                break;
-            // Submit the JCL from a local file
-            case "local-file": {
-                parms.jclSource = this.mArguments.localFile;
-                const JclString = fs.readFileSync(this.mArguments.localFile).toString();
-                apiObj = await SubmitJobs.submitJclString(this.mSession, JclString, parms);
-                source = this.mArguments.localFile;
-                if (parms.viewAllSpoolContent) {
-                    spoolFilesResponse = apiObj;
+                    break;
+                // Submit the JCL from a local file
+                case "local-file": {
+                    parms.jclSource = this.mArguments.localFile;
+                    try {
+                        const JclString = fs.readFileSync(this.mArguments.localFile).toString();
+                        apiObj = await SubmitJobs.submitJclString(this.mSession, JclString, parms);
+                        source = this.mArguments.localFile;
+                        if (parms.viewAllSpoolContent) {
+                            spoolFilesResponse = apiObj;
+                        }
+                    } catch (err) {
+                        throw new ImperativeError({msg: err});
+                    }
```

Comment:
If `SubmitJobs.submitJclString` throws an error, then "recasting" the error in the catch block may lose some useful error details. So I don't think we want a "blanket" try-catch here.

Instead you might want to wrap only the first line in a try-catch, to handle if the `fs.readFileSync` call throws an I/O error. This would match the error handling of the `files upload` command that Gene mentioned 😋 
```
try {
    const JclString = fs.readFileSync(this.mArguments.localFile).toString();
} catch (err) {
    throw new ImperativeError({
        msg: ZosFilesMessages.nodeJsFsError.message,
        additionalDetails: err.toString(),
        causeErrors: err
    });
}
```

Note: We'll need to copy over the "nodeJsFsError" message from ZosFilesMessages to ZosJobsMessages if we want to use it here in the z/OS Jobs SDK 😋